array_api_compat/torch/_aliases.py

```diff
@@ -681,7 +681,11 @@ def isdtype(
     else:
         return dtype == kind
 
-def take(x: array, indices: array, /, *, axis: int, **kwargs) -> array:
+def take(x: array, indices: array, /, *, axis: Optional[int] = None, **kwargs) -> array:
+    if axis is None:
+        if x.ndim != 1:
+            raise ValueError("axis must be specified when ndim > 1")
+        axis = 0
```

Comment:
Or you can dispatch to `torch.take`, either should be alright.

Reply:
torch.take doesn't support an axis argument.

Reply:
exactly :D `torch.take() == numpy.take(axis=None)`

Reply:
it's a bit more general, as it sees the whole tensor as a 0dim tensor and indexes into it, so it also works for `ndim > 1`.

Reply:
Right. The flattening behavior is also there in NumPy, but I think we wanted to avoid that with the array API `take`. I'm not sure to what degree we should try to avoid that for the compat library, though (c.f. https://github.com/data-apis/array-api-compat/issues/34#issuecomment-1636813887)